### PR TITLE
Create recipe for OpenEXR

### LINKED ISSUE #1802: Add OpenEXR
It would be great to have a package for OpenEXR.

cc @patricksnape

### PR DESCRIPTION
**Pull request by aaronang**
This is my first time creating a conda package. I am looking forward to receiving feedback 😁 

Resolves #1802.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
Woot!
I will be giving some feedback later in the day!

File: recipes/openexr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+    - {{ pin_compatible('python', lower_bound='2.7') }}
```
Comment (jakirkham) [inline recipes/openexr/meta.yaml]: What's the `pin_compatible` here for? If the goal is to just ensure `python` is in the build environment and is at least `2.7`, then listing `python` should be sufficient. If you want to force Python `2.7`, then you can add `python 2.7.*`.

File: recipes/openexr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - boost
+  host:
+    - zlib
+  run:
+    - zlib
+
+test:
+  commands:
+    - test -e $PREFIX/include/OpenEXR/OpenEXRConfig.h  # [unix]
+    - test -e $PREFIX/include/OpenEXR/IlmBaseConfig.h  # [unix]
```
Comment (jakirkham) [inline recipes/openexr/meta.yaml]: Should we add some file tests for libraries installed here?

File: recipes/openexr/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+cd build
+
+cmake $SRC_DIR -G "Ninja" \
+    -DCMAKE_BUILD_TYPE:STRING=Release \
+    -DCMAKE_PREFIX_PATH:PATH=$PREFIX \
+    -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX \
+    -DOPENEXR_BUILD_PYTHON_LIBS:BOOL=OFF \
+    -DOPENEXR_NAMESPACE_VERSIONING:BOOL=OFF \
+    -DOPENEXR_BUILD_STATIC:BOOL=ON
+
```
Comment (jakirkham) [inline recipes/openexr/build.sh]: Is there a `ninja check` we can use to run the tests here?

**Comment by jakirkham** [comment]
Thanks for working on this @aaronang 😄 Added a couple of comments above. Generally looks like it is in pretty good shape.

**Comment by jakirkham** [comment]
Thanks @aaronang.

Looks like the build target is called `test` instead. Sorry wasn't sure which target name they would choose. :)

**Comment by SylvainCorlay** [comment]
This looks really good. Is there any reason to skip windows?

The QuantForge recipe built succesfully on windows, I guess this should also work here:

https://github.com/QuantForge/openexr-recipe/blob/master/recipe/bld.bat

**Comment by aaronang** [comment]
@SylvainCorlay I just don't have much experience with Windows 😛 

@jakirkham Will try it out 👍

**Comment by SylvainCorlay** [comment]
@aaronang I think that replacing `skip: True  # [win]` with `skip: True  # [win and vc<14]` plus taking the bld.bat that I pointed you to should work. The QuantForge recipe worked without an issue.

You probably want to add pass the same cmake options as for the linux build.

**Comment by SylvainCorlay** [comment]
PS @aaronang if you make more change to the recipe, feel free to add me as a maintainer.

With the compiler migration is done, I will be maintaining several recipes depending on openexr.

File: recipes/openexr/bld.bat [hunk 4]
```diff
@@ -0,0 +1,8 @@
+cmake -G "NMake Makefiles" -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% -DILMBASE_PACKAGE_PREFIX=%LIBRARY_PREFIX% %SRC_DIR%
```
Comment (SylvainCorlay) [inline recipes/openexr/bld.bat]: Maybe we need `-DOPENEXR_BUILD_PYTHON_LIBS:BOOL=OFF` for the windows build.
Other options that are passed on unix and not windows are: `-DOPENEXR_NAMESPACE_VERSIONING:BOOL=OFF` and `-DOPENEXR_BUILD_STATIC:BOOL=ON`

**Comment by SylvainCorlay** [comment]
It seems that you need to pass `-DOPENEXR_BUILD_TESTS` for the tests to be built, or remove the `ninja test` section.

**Comment by SylvainCorlay** [comment]
Opened #7088 adding the -DOPENEXR_BUILD_TESTS and run_exports section.

**Comment by aaronang** [comment]
Closing due to #7088. Thank you @SylvainCorlay 🙇🙇🙇

**Pull request closed without merging**